pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -489,11 +492,124 @@ func describePodInServiceGroup(f formatter, podNode *kubegraph.PodNode) []string
 	return lines
 }
 
+type exposedRoutes []string
+
+func (e exposedRoutes) Len() int      { return len(e) }
+func (e exposedRoutes) Swap(i, j int) { e[i], e[j] = e[j], e[i] }
+func (e exposedRoutes) Less(i, j int) bool {
+	a, b := e[i], e[j]
+	httpA, httpB := strings.HasPrefix(a, "https://"), strings.HasPrefix(b, "https://")
+	switch {
+	case httpA && !httpB:
+		return true
+	case !httpA && httpB:
+		return false
+	case !httpA && !httpB:
+		httpA, httpB = strings.HasPrefix(a, "http://"), strings.HasPrefix(b, "http://")
+		switch {
+		case httpA && !httpB:
+			return true
+		case !httpA && httpB:
+			return false
+		case !httpA && !httpB:
+			return a < b
+		default:
+			a, b = a[8:], b[8:]
+		}
+	default:
+		a, b = a[7:], b[7:]
+	}
+	lA, lB := strings.Index(a, " "), strings.Index(b, " ")
+	if lA == -1 {
+		lA = len(a)
+	}
+	if lB == -1 {
+		lB = len(b)
+	}
+	switch {
+	case lA < lB:
+		return true
+	case lA > lB:
+		return false
+	default:
+		return a < b
+	}
+}
+
+func extractRouteInfo(route *routeapi.Route) (requested bool, other []string, errors []string) {
+	reasons := sets.NewString()
+	for _, ingress := range route.Status.Ingress {
+		exact := route.Spec.Host == ingress.Host
+		switch status, condition := routeapi.IngressConditionStatus(&ingress, routeapi.RouteAdmitted); status {
+		case kapi.ConditionFalse:
+			reasons.Insert(condition.Reason)
+		default:
+			if exact {
+				requested = true
+			} else {
+				other = append(other, ingress.Host)
+			}
+		}
+	}
+	return requested, other, reasons.List()
+}
+
+func describeRouteExposed(host string, route *routeapi.Route, errors bool) string {
+	var trailer string
+	if errors {
+		trailer = " (!)"
+	}
+	var prefix string
+	switch {
+	case route.Spec.TLS == nil:
+		prefix = fmt.Sprintf("http://%s", host)
+	case route.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough:
+		prefix = fmt.Sprintf("https://%s (passthrough)", host)
+	case route.Spec.TLS.Termination == routeapi.TLSTerminationReencrypt:
+		prefix = fmt.Sprintf("https://%s (reencrypt)", host)
+	case route.Spec.TLS.Termination != routeapi.TLSTerminationEdge:
```

Comment:
I think this case is redundant. If tls is specified then, it has to be one of the three types.